.github/workflows/main.yaml

```diff
@@ -257,9 +267,17 @@ jobs:
         uses: ./.github/actions/gn
       - name: Build Cobalt
         uses: ./.github/actions/build
+      - name: 'Check Artifact Size'
+        uses: ./.github/actions/check_artifact_size
+        if: ${{ (inputs.check_artifact_size && needs.initialize.outputs.evergreen_loader == 'null') || (inputs.check_artifact_size && needs.initialize.outputs.evergreen_loader != 'null' && matrix.sb_api_version == '') }}
+        with:
+          workflow: ${{ inputs.check_artifact_size }}
+          name: ${{ matrix.platform }}-${{ matrix.config }}
+          path: out/${{ matrix.platform }}_${{ matrix.config }}/${{ inputs.keep_artifacts }}
+          threshold: ${{ inputs.artifact_size_increase_percentage_threshold }}
       - name: 'Upload Artifact'
         uses: actions/upload-artifact@v4
-        if: ${{ inputs.keep_artifacts && matrix.sb_api_version == '' }}
+        if: ${{ (inputs.keep_artifacts && needs.initialize.outputs.evergreen_loader == 'null') || (inputs.keep_artifacts && needs.initialize.outputs.evergreen_loader != 'null' && matrix.sb_api_version == '') }}
```

Comment:
This is becoming hard to read. Can we factor these extra conditions out to a toplevel variable ? 

`keep_artifacts` denotes the list of artifacts to keep ( and perhaps this could be better named ), and the other conditions seem to be extra layers that resolve to a `boolean` of whether these listed artifacts should be uploaded or not ( e.g. `upload_artifacts` ). That boolean can be evaluated in a separate step with `outputs:` defined.